Internal: Hide component from components tab [ED-21261]

MuliElementor opened the pull request:
## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

<!--start_gitstream_placeholder-->
### ✨ PR Description
```
Purpose: Add archive functionality to components to enable hiding them from the components tab while maintaining their data in the system.
Main changes:
- Added archive/unarchive capability to components with REST API endpoints and data persistence
- Implemented UI for archiving components through a context menu in component items
- Added filtering of archived components from component list display
- Created pre-save synchronization to update archived component status on the server
```

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>
<!--end_gitstream_placeholder-->

MuliElementor committed:
Implement component archiving functionality in the REST API

- Added a new endpoint to archive components via a POST request.
- Introduced an `archive` method in the Components_Repository to handle archiving logic.
- Updated the components list to filter out archived components.
- Created a custom hook `useArchive` to manage the archiving process in the UI.
- Enhanced the ComponentItem component to include an archive action in the menu.

This update improves component management by allowing users to archive components, enhancing the overall user experience.

MuliElementor committed:
Enhance component archiving functionality to support multiple IDs

- Updated the `archive` method in `Components_Repository` to accept an array of component IDs for archiving.
- Modified the REST API to handle multiple component IDs in the `archive_component` endpoint.
- Renamed the `archiveComponent` method in the API client to `updateArchivedComponents` for clarity.
- Adjusted the `useArchive` hook to work with multiple component IDs.
- Enhanced the `ComponentItem` component to trigger archiving for individual components while maintaining document modification status.

These changes improve the flexibility and usability of the component archiving feature.

MuliElementor committed:
Refactor component archiving methods for consistency and clarity

- Renamed the `archive_component` method to `archive_components` in the Components_REST_API class to reflect the handling of multiple component IDs.
- Updated the callback in the REST API route to use the new method name.
- Adjusted error logging and response messages to align with the new method name.
- Modified the API client to correctly handle the response structure from the archiving request.

These changes enhance the clarity and maintainability of the component archiving functionality.

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

MuliElementor committed:
Enhancement: Refactor component archiving logic to improve error handling and response structure. Updated the `archive` method to return detailed success and failure IDs. Introduced `get_is_archived` method for better state management. Updated REST API and frontend to accommodate new response format.

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

MuliElementor committed:
Enhancement: Add accessibility improvements to animation play button with aria-label and implement end-to-end tests for animation functionality in the Elementor editor. This includes verifying animation event firing and rendering completion.

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

MuliElementor committed:
format

MuliElementor committed:
format

MuliElementor committed:
update

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

MuliElementor committed:
update

MuliElementor committed:
update tests

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

MuliElementor committed:
update and create tests

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

MiryamOren commented on packages/packages/core/editor-components/src/components/__tests__/component-item.test.tsx:
```diff
@@ -0,0 +1,71 @@
+		store = __createStore();
+	} );
+	it( 'should render', () => {
+		dispatch( slice.actions.load( mockComponents ) );
```
please add // Arrange // Act // Assert here too

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

elementorbot committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

IshayMaya commented on packages/packages/core/editor-components/src/hooks/use-archive.ts:
```diff
@@ -0,0 +1,15 @@
+		},
+		[ dispatch ]
+	);
+};
```
i think it can just be an action, not a hook (will also be more helpful for mcp)

```suggestion
export const archiveComponent = ( componentId: number ) => {
	const dispatch = getStore()?.dispatch;

	if ( ! dispatch ) {
		return;
	}

	dispatch( slice.actions.archive( componentId ) );
};
```

IshayMaya commented on packages/packages/core/editor-components/src/components/components-tab/components-item.tsx:
```diff
@@ -26,27 +47,55 @@ export const ComponentItem = ( { component }: ComponentItemProps ) => {
+				} }
+			>
+				<MenuListItem sx={ { minWidth: '160px' } } onClick={ handleArchiveClick }>
+					<Typography variant="caption" sx={ { color: 'text.primary' } }>
```
didn't we already solve the customization in the editor-ui component?

IshayMaya commented on packages/packages/core/editor-components/src/sync/update-archived-component-before-save.ts:
```diff
@@ -0,0 +1,40 @@
+
+		if ( result.successArchivedIds.length > 0 ) {
+			notify( successNotification( result.successArchivedIds.join( ', ' ) ) );
+		}
```
```suggestion
		const result = await apiClient.updateArchivedComponents(
			archivedComponents.map( ( component ) => component.id )
		);

		const failedArchivedIds = result.failedArchivedIds.join( ', ' );
		const successArchivedIds = result.successArchivedIds.join( ', ' );

		if ( failedArchivedIds ) {
			notify( failedNotification( failedArchivedIds ) );
		}

		if ( successArchivedIds ) {
			notify( successNotification( successArchivedIds ) );
		}
```

IshayMaya commented on packages/packages/core/editor-components/src/components/__tests__/component-item.test.tsx:
```diff
@@ -0,0 +1,71 @@
+import { SearchProvider } from '../components-tab/search-provider';
+
+jest.mock( '@elementor/editor-documents', () => ( {
+	setDocumentModifiedStatus: jest.fn(),
```
lets also test that it was called

MuliElementor committed:
Refactor: Update archive handling in components repository and API

- Renamed variables in the `archive` method of `Components_Repository` for clarity.
- Updated API client to reflect new variable names for archived component responses.
- Refactored `use-archive` hook to streamline component archiving and document modification status.
- Adjusted tests to ensure proper functionality of component archiving and state management.

This change enhances code readability and maintains consistency across the component archiving process.

MuliElementor committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

MuliElementor committed:
Refactor: Move archiveComponent to store actions and remove use-archive hook

- Moved the `archiveComponent` function from the `use-archive` hook to `store/actions.ts` for better organization.
- Deleted the now redundant `use-archive.ts` file.
- Updated imports in `components-item.tsx` to reflect the new location of `archiveComponent`.
- Enhanced state management by integrating document modification status directly within the action.

This change improves code structure and maintains consistency in component archiving functionality.

MuliElementor committed:
update tests

MuliElementor committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab

# Conflicts:
#	packages/packages/core/editor-components/package.json

MuliElementor committed:
update

MuliElementor committed:
update

MuliElementor committed:
update

MuliElementor committed:
update

MuliElementor committed:
update

MuliElementor committed:
add editor-norifications

MuliElementor committed:
update

MuliElementor committed:
Merge branch 'main' into ED-21261-hide-component-from-components-tab
